tst/EnergyPlus/unit/SingleDuct.unit.cc

```diff
@@ -2835,3 +2835,181 @@ TEST_F(EnergyPlusFixture, setATMixerSizingProperties_Test)
+        "    Zone 1 TU NoReheat,       !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    Zone 1 TU Air Inlet,     !- Air Inlet Node Name",
+        "    Zone 1 Supply Inlet,     !- Air Outlet Node Name",
+        "    1.0;                     !- Maximum Air Flow Rate {m3/s}",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool ErrorsFound = false;
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+    DataZoneEquipment::GetZoneEquipmentData(*state);
+    state->dataSize->TermUnitFinalZoneSizing.allocate(1);
+    state->dataSize->TermUnitSizing.allocate(1);
+    state->dataSize->FinalZoneSizing.allocate(1);
+    ZoneAirLoopEquipmentManager::GetZoneAirLoopEquipment(*state);
+    SingleDuct::GetSysInput(*state);
+    EXPECT_TRUE(compare_err_stream(""));
+
+    int SysNum = 1;
+
+    // test air flow sizing of TU with hard sized input for Maximum Air Flow Rate {m3/s}
+    state->dataSize->ZoneSizingRunDone = true;
+    state->dataSize->CurZoneEqNum = 1;
+    state->dataSize->CurTermUnitSizingNum = 1;
+    state->dataSize->TermUnitFinalZoneSizing(1).NonAirSysDesHeatVolFlow = 0.5;
+    state->dataSingleDuct->sd_airterminal(SysNum).SizeSys(*state);
+    EXPECT_EQ(0.0, state->dataSingleDuct->sd_airterminal(SysNum).ZoneMinAirFracDes);
+    // TermUnitSizing(1).AirVolFlow should be greater of user input Maximum Air Flow Rate (1.0) and NonAirSysDesHeatVolFlow (0.5)
+    EXPECT_EQ(1.0, state->dataSize->TermUnitSizing(1).AirVolFlow);
+    // MaxAirVolFlowRate hard sized at 1.0
+    EXPECT_EQ(1.0, state->dataSingleDuct->sd_airterminal(SysNum).MaxAirVolFlowRate);
+
+    // MaxAirVolFlowRate autosized
+    state->dataSingleDuct->sd_airterminal(SysNum).MaxAirVolFlowRate = DataSizing::AutoSize;
+    state->dataSize->TermUnitFinalZoneSizing(1).DesHeatVolFlow = 1.0;
+    state->dataSingleDuct->sd_airterminal(SysNum).SizeSys(*state);
+    // constant volume TUs do not set or use ZoneMinAirFracDes
+    EXPECT_EQ(0.0, state->dataSingleDuct->sd_airterminal(SysNum).ZoneMinAirFracDes);
+    // TermUnitSizing(1).AirVolFlow should be greater of DesHeatVolFlow (1.0) and NonAirSysDesHeatVolFlow (0.5)
+    EXPECT_EQ(1.0, state->dataSize->TermUnitSizing(1).AirVolFlow); // doesn't matter for NoReheat TUs
+    EXPECT_EQ(1.0, state->dataSingleDuct->sd_airterminal(SysNum).MaxAirVolFlowRate);
+
+    state->dataSingleDuct->sd_airterminal(SysNum).MaxAirVolFlowRate = DataSizing::AutoSize;
+    state->dataSize->TermUnitFinalZoneSizing(1).DesHeatVolFlow = 0.25;
+    state->dataSingleDuct->sd_airterminal(SysNum).SizeSys(*state);
+    // TermUnitSizing(1).AirVolFlow should be greater of DesHeatVolFlow (0.25) and NonAirSysDesHeatVolFlow (0.5)
+    EXPECT_EQ(0.5, state->dataSize->TermUnitSizing(1).AirVolFlow);
+    // the TU will size to DesHeatVolFlow (0.25) even though NonAirSysDesHeatVolFlow = 0.5
+    EXPECT_EQ(0.25, state->dataSingleDuct->sd_airterminal(SysNum).MaxAirVolFlowRate);
```

Comment:
These last 2 tests here are bothering me. TU air flow (sd_airterminal(SysNum).MaxAirVolFlowRate) sizes to greater of TUFZS cooling or heating (0.25). 

    MaxAirVolFlowRateDes = max(state.dataSize->TermUnitFinalZoneSizing(state.dataSize->CurTermUnitSizingNum).DesCoolVolFlow,
                               state.dataSize->TermUnitFinalZoneSizing(state.dataSize->CurTermUnitSizingNum).DesHeatVolFlow);

and TermUnitSizing.AirVolFlow sizes to greater of that and NonAirSysDesHeatVolFlow (0.5). Feels like an ill formed unit test or a logic problem in sizing. Darn these sizing changes. I hope that this is caught in setting TermUnitFinalZoneSizing air flow.

                TermUnitSizing(state.dataSize->CurTermUnitSizingNum).AirVolFlow =
                    max(state.dataSize->TermUnitFinalZoneSizing(state.dataSize->CurTermUnitSizingNum).NonAirSysDesHeatVolFlow,
                        this->MaxAirVolFlowRate * this->ZoneTurndownMinAirFrac);

Reply:
NonAirSysDesHeatVolFlow is the raw zone heating air flow set in EndZoneSizingCalc

    z.NonAirSysDesHeatVolFlow = c.DesHeatVolFlow;

And then later in that function is multiplied by the heating sizing factor.

    // update non air system design load and air flow to include the sizing factor
    state.dataSize->FinalZoneSizing(CtrlZoneNum).NonAirSysDesHeatLoad *= state.dataSize->FinalZoneSizing(CtrlZoneNum).HeatSizingFactor;
    state.dataSize->FinalZoneSizing(CtrlZoneNum).NonAirSysDesHeatVolFlow *= state.dataSize->FinalZoneSizing(CtrlZoneNum).HeatSizingFactor;

which could be < 1

    Sizing:Zone,
    N7, \field Zone Heating Sizing Factor
         \note if blank or zero, global heating sizing factor from Sizing:Parameters is used.
         \minimum 0

    Sizing:Parameters,
      N1, \field Heating Sizing Factor,
          \type real
          \minimum> 0.0
          \default 1.0

So it appears it is possible that FinalZoneSizing.NonAirSysDesHeatVolFlow could be less than FinalZoneSizing.DesHeatVolFlow, at this point in the code. DesHeatVolFlow is multiplied by HeatSizingFactor somewhere in this mix I just don't see it right now.

Then in UpdateTermUnitFinalZoneSizing the TermUnitSizing array is set. 

        auto &thisTUFZSizing(state.dataSize->TermUnitFinalZoneSizing(termUnitSizingIndex));
        auto &thisTUSizing(state.dataSize->TermUnitSizing(termUnitSizingIndex));
        // Copy everything from FinalZoneSizing to TermUnitFinalZoneSizing
        thisTUFZSizing = thisFZSizing;

To test this I used 8713-5ZoneAirCooledConvCoef-Modified

    Zone Sizing Information, SPACE5-1, Cooling, 2552.94399, 3318.82718, 0.21475, 0.27917, CHICAGO ANN CLG 1% CONDNS DB=>MWB, 7/21 15:00:00, 31.50000, 1.46163E-002, 182.49000, 20.00000, 0.18880, 0.00000
    Zone Sizing Information, SPACE5-1, Heating, 1690.05388, 2197.07005, 4.91278E-002, 0.18880, CHICAGO ANN HTG 99% CONDNS DB, 1/21 24:00:00, -17.30000, 8.38927E-004, 182.49000, 20.00000, 0.18880, 0.00000
    Component Sizing Information, AirTerminal:SingleDuct:ConstantVolume:Reheat, SPACE5-1 CV REHEAT, Design Size Maximum Air Flow Rate [m3/s], 0.27917
    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size Rated Capacity [W], 4952.51303

and then changed sizing factors of 1.3 by exactly 1/2 to 0.65 expecting TU flow to reduce 0.27917 by 1/2

    Sizing:Parameters,
      0.65,                     !- Heating Sizing Factor
      0.65,                     !- Cooling Sizing Factor
      ;                        !- Timesteps in Averaging Window

I also have to cut the DesignSpecification:OutdoorAir for zone 5 in half (0.00944 per person to 0.00472) so that zone OA flow is not dominant and I do get what I expect. Air flow and heating coil capacity are 1/2 of what they were before.

    Zone Sizing Information, SPACE5-1, Cooling, 2552.94399, 1659.41359, 0.21475, 0.13959, CHICAGO ANN CLG 1% CONDNS DB=>MWB, 7/21 15:00:00, 31.50000, 1.46163E-002, 182.49000, 20.00000, 9.44000E-002, 0.00000
    Zone Sizing Information, SPACE5-1, Heating, 1690.05388, 1098.53502, 4.91278E-002, 9.44000E-002, CHICAGO ANN HTG 99% CONDNS DB, 1/21 24:00:00, -17.30000, 8.38927E-004, 182.49000, 20.00000, 9.44000E-002, 0.00000
    Component Sizing Information, AirTerminal:SingleDuct:ConstantVolume:Reheat, SPACE5-1 CV REHEAT, Design Size Maximum Air Flow Rate [m3/s], 0.13959
    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size Rated Capacity [W], 2476.25651

This PRs code change seems OK for now. The unit test is doing something that apparently can't happen in code.